### PR TITLE
Add unit tests to map component.

### PR DESCRIPTION
**Pull request by DaoyuT**
Add unit tests to map component.

### CODE REVIEW

**Comment by gino-m** [comment]
@DaoyuT Something seems to be failing with:
```
ERROR: 'InvalidValueError: setMap: not an instance of Map; and not an instance of StreetViewPanorama'
```

File: web-ng/src/app/components/map/map.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import { GoogleMapsModule } from '@angular/google-maps';
+
+// import { initialize } from "@googlemaps/jest-mocks";
+// import createGoogleMapsMock from 'jest-google-maps-mock';
+
+describe('MapComponent', () => {
+  let component: MapComponent;
+  let fixture: ComponentFixture<MapComponent>;
+  const dialog: Partial<MatDialog> = {};
+  const map: jasmine.SpyObj<google.maps.Map> = jasmine.createSpyObj(
```
Comment (gino-m) [inline web-ng/src/app/components/map/map.component.spec.ts:39]: IIuc, this uses as real Map instance but stubs out selected methods. This still makes the test dependent on the impl of Map. Can we use a fake here instead, declaring stubs only for the methods we use?

File: web-ng/karma.conf.js [hunk 2]
```diff
@@ -54,5 +54,15 @@ module.exports = function (config) {
         flags: ['--no-sandbox'],
       },
     },
+    // files: ['src/app/shared/mocks/google-maps-api.js'],
```
Comment (gino-m) [inline web-ng/karma.conf.js:57]: Remove commented lines

File: web-ng/src/app/app.component.ts [hunk 3]
```diff
@@ -29,8 +29,11 @@ import { environment } from '../environments/environment';
 })
 export class AppComponent {
   public constructor(@Inject(DOCUMENT) private doc: Document) {
-    this.initGoogleMap();
+    if (environment.googleMapsApiKey !== '') {
```
Comment (gino-m) [inline web-ng/src/app/app.component.ts:32]: Please add a comment indicating this is for testing

**Review by gino-m — CHANGES_REQUESTED** [review]
Nice work, pretty good coverage of the map component!

@os-micmec Could you also PTAL?

File: web-ng/src/app/app.component.ts (hunk 3 above)
Comment (DaoyuT) [reply web-ng/src/app/app.component.ts]: Added an `env` entry in environment variables, and used that.

**Review by gino-m — DISMISSED** [review]

**Pull request merged**